Assets/GameAssets/Player/PlayerScripts/Harpoon/HarpoonGun.cs

```diff
@@ -275,14 +294,40 @@ private void HarpoonFullyReloaded()
     {
         _harpoonFiringState = EHarpoonFiringState.Ready;
         _harpoonOnGun.SetActive(true);
+        RuntimeSfxManager.APlayOneShotSfx?
+            .Invoke(FmodSfxEvents.Instance.HarpoonReload, gameObject.transform.position);
 
-        if(_isFocusButtonHeld)
+        if (_isFocusButtonHeld)
         {
             StartFocusingHarpoon();
         }
 
         PlayerManager.Instance.InvokeOnHarpoonReloadedEvent();
     }
+
+    /// <summary>
+    /// Restocks harpoons to full from the given ammo rack
+    /// </summary>
+    /// <param name="ammoRack"> The ammo rack to take harpoons from </param>
+    private void RestockHarpoons(AmmoRackInteractable ammoRack)
+    {
+        int targetAmmo = 0;
+
+        if (_harpoonFiringState != EHarpoonFiringState.Ready)
+        {
+            targetAmmo = _maxAmmo;
```

Comment:
If the above comment of reducing ammo when firing is addressed then we shouldn't need the if statement with 2 different ammo counts

Reply:
See above comment of reducing ammo when firing.